src/typechecker.rs

```diff
@@ -1844,6 +1847,53 @@ fn typecheck_function(
     error
 }
 
+fn typecheck_jakt_main(function: &ParsedFunction) -> Option<JaktError> {
+    let param_type_error = Some(JaktError::TypecheckError(
+        "Main function must take a single array of strings as its parameter".to_string(),
+        function.name_span,
+    ));
+
+    if function.params.len() > 1 {
+        return param_type_error;
+    }
+
+    if !function.params.is_empty() {
+        match &function.params[0].variable.ty {
```

Comment:
Just noticed I missed this in my `ty` removal & renamed it now, please rebase/update

Reply:
Is it okay though to use it in 
`ParsedType::Array(ty, _) => match &**ty`
or should these destructurings also have more verbose names?

Reply:
it's not ok to use anywhere :)